### PR TITLE
gh-112535: Add comment for ppc32/64 registers

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-112535

### CODE REVIEW

**Comment by corona10** [comment]
Reference: 
32bit: https://github.com/ryanarn/powerabi
64bit: https://ftp.libre-soc.org/PPC-elf64abi-1.4.1.txt

File: Include/object.h [hunk 1]
```diff
@@ -265,6 +265,7 @@ _Py_ThreadId(void)
     #if defined(__clang__) && _Py__has_builtin(__builtin_thread_pointer)
     tid = (uintptr_t)__builtin_thread_pointer();
     #else
+    // r13 is reserved for use as system thread ID by the 64-bit ABI.
```
Comment (vstinner) [inline Include/object.h]:
```suggestion
    // r13 is reserved for use as system thread ID by the Power 64-bit ABI.
```

File: Include/object.h [hunk 2]
```diff
@@ -273,6 +274,7 @@ _Py_ThreadId(void)
     #if defined(__clang__) && _Py__has_builtin(__builtin_thread_pointer)
     tid = (uintptr_t)__builtin_thread_pointer();
     #else
+    // r2 is reserved for use as system thread ID by the 32-bit ABI.
```
Comment (vstinner) [inline Include/object.h]:
```suggestion
    // r2 is reserved for use as system thread ID by the Power 32-bit ABI.
```

**Review by vstinner — APPROVED** [review]
LGTM. Thanks for these adding these comments :-)

**Pull request merged**